lib/src/over_react_test/react_util.dart

```diff
@@ -481,8 +491,35 @@ Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey
 ///     </div>
 ///
 ///     queryAllByTestId(renderedInstance, 'value'); // returns both `inner` `<div>`s
-List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey}) {
-  return findDomNode(root).querySelectorAll('[$key~="$value"]');
+List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int shadowDepth}) {
+  return _findDeep(findDomNode(root), _makeTestIdSelector(value, key: key), searchInShadowDom: searchInShadowDom, findMany: true, depth: shadowDepth);
+}
+
+String _makeTestIdSelector(String value, {String key = defaultTestIdKey}) => '[$key~="$value"]';
+
+List<Element> _findDeep(Node root, String itemSelector, {bool searchInShadowDom = false, bool findMany = true, int depth}) {
+  List<Element> nodes = [];
+  final rootQuerySelector = root is ShadowRoot ? root.querySelector : root is Element ? root.querySelector : null;
+  var lightElement = rootQuerySelector(itemSelector);
+  if (!findMany && lightElement != null){
+    return [lightElement];
+  }
+  void recursiveSeek(Node _root, int _currentDepth) {
+    final List<Element> Function(String) rootQuerySelectorAll = _root is ShadowRoot ? _root.querySelectorAll : _root is Element ? _root.querySelectorAll : null;
```

Comment:
#nit unnecessary LHS typing
```suggestion
    final rootQuerySelectorAll = _root is ShadowRoot ? _root.querySelectorAll : _root is Element ? _root.querySelectorAll : null;
```

Reply:
running `ddev test` 
Without LHS typing: `00:04 +358 -9: Some tests failed`
With LHS typing: `00:04 +367: All tests passed!`

Reply:
Wait, so adding this typing alters runtime behavior? That's really strange.  What's the cause of the failure?

If this is needed, I'm fine with that, but it could use a comment.

Reply:
what's weird and this was actually what fixed the issue that i asked for your help on. like when you dont do this typing, it doesn't throw or fail it just like doesn't work. every test will be like nope there were no elements in the list from rootQuerySelectorAll....

Reply:
LOL JK... so it throws `Type '_FrozenElementList<JSObject<undefined>>' should be '_FrozenElementList<Element>'` if you don't have the `.toList()` below but then upon adding the `.toList()` it does what i mention above.

Reply:
Gotcha; so could we just add a comment saying something along the lines of "the LHS type prevents this from returning `_FrozenElementList<JSObject<undefined>>` instead of `<Element>` in DDC"?

Reply:
thumbs